src/base/decoder/baseDecoder.ts

```diff
@@ -0,0 +1,33 @@
+import { ContractABI } from '../iface';
+import { BaseFunctionCallExplanation, BaseMethodIdMapping } from './iface';
+
+// import { MethodContainerMap, MethodManager } from '../method/manager';
+
+/**
+ * A high-level wrapper for Solidity smart contract function calls
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export abstract class BaseDecoder< K extends BaseMethodIdMapping, T extends BaseFunctionCallExplanation, S extends ContractABI > {
```

Comment:
these `K, T, S` variables should be interfaces, not concrete implementations, it's kind of backwards here

you want the classes at a higher level in the eth decoder and then in the base you want interfaces. this leave some flexibility for other classes that want to implement `BaseDecoder` as another abstract class